Mathlib/Algebra/MvPolynomial/SchwartzZippel.lean

```diff
@@ -0,0 +1,145 @@
+/-
+Copyright (c) 2023 Bolton Bailey. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bolton Bailey, Yaël Dillies
+-/
+import Mathlib.Algebra.MvPolynomial.Equiv
+import Mathlib.Algebra.MvPolynomial.Variables
+import Mathlib.Algebra.Polynomial.Roots
+import Mathlib.Data.Fin.Tuple.Finset
+
+/-!
+# The Schwartz-Zippel lemma
+
+This file contains a proof of the
+[Schwartz-Zippel](https://en.wikipedia.org/wiki/Schwartz%E2%80%93Zippel_lemma) lemma.
+
+This lemma tells us that the probability that a nonzero multivariable polynomial over an integral
+domain evaluates to zero at a random point is bounded by the degree of the polynomial over the size
+of the field, or more generally, that a nonzero multivariable polynomial over any integral domain
+has a low probability of being zero when evaluated at points drawn at random from some finite subset
+of the field. This lemma is useful as a probabilistic polynomial identity test.
+
+## TODO
+
+* Generalize to polynomials over arbitrary variable types
+* Write a tactic to apply this lemma to a given polynomial
+* Can the RHS be strengthened to `p.support.sup fun s ↦  ∑ i, s i / #S i`? Namely, can the sup
+  inside the `p.degreeOf` be pulled outside?
```

Comment:
Isn't it already done?

Reply:
No, I have currently proved `_ ≤ ∑ i, (p.support.sup fun s ↦ s i) / #S i`, not `_ ≤ p.support.sup fun s ↦ ∑ i, s i / #S i`

Reply:
Update: It's done.